contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/DrillHiveMetaStoreClient.java

```diff
@@ -209,8 +213,10 @@ private DrillHiveMetaStoreClient(final HiveConf hiveConf) throws MetaException {
       throws TException {
     try {
       return mClient.getAllTables(dbName);
+    } catch (MetaException | UnknownDBException e) {
+      throw e;
     } catch (TException e) {
-      logger.warn("Failure while attempting to get hive tables", e);
+      logger.warn("Failure while attempting to get hive tables. Retries once.", e);
```

Comment:
Add code to check if close() hits exception. In case close() fails with exception, we log the exception and continue to reconnect(). This may leak connection. 

However, we have to maintain hive metastore connection. Otherwise, next hive query may fail, and the only way to continue is either re-enable hive storage plugin (which may then leak a connection DRILL-4774), or re-start drillbit.